### DISCUSSION TITLE
Twitter 3-legged OAuth

### DISCUSSION THREAD

**Discussion by bigansh** [plhery/node-twitter-api-v2#267] [Q&A] [answered]
Do we get the user's details (like email, photo-URL, name, etc.) once the user approves the app?

**Comment by alkihis** [comment, marked as answer]
Most of user account details can be found in `.v1.currentUser()` (only if you've used OAuth1, see endpoint `/1.1/account/verify_credentials.json`) or `.v2.me()` with the expansions you need (see `/2/users/me` endpoint)

**Reply by bigansh** [reply]
Got it. Thanks.

**Reply by bigansh** [reply]
Hey, if I want to access the email id, what should I do? I have passed several scopes likes `[
				'users.read',
				'bookmark.read',
				'bookmark.write',
				'offline.access',
			]`; but I am not able to get the email id.

**Reply by adxmcollins** [reply]
You might've figured this out since, but I came across the same issue.

It looks like you have to use `v1.verifyCredentials({ include_email: true })` instead of `currentUser()` if you want their email address. The response appears to be the same just with the email address so I assume it hits the same endpoint on Twitter.
